gateway/src/apicast/policy/nginx_metrics/nginx_metrics.lua

```diff
@@ -118,19 +126,22 @@ function _M:metrics()
   end
 end
 
-local function report_req_response_time()
+local function report_req_response_time(service)
   -- Use ngx.var.original_request_time instead of ngx.var.request_time so
   -- the time spent in the post_action phase is not taken into account.
   local resp_time = tonumber(ngx.var.original_request_time)
-
   if resp_time and response_times then
-    response_times:observe(resp_time)
+    response_times:observe(resp_time, { service })
   end
 end
 
-function _M.log()
-  upstream_metrics.report(ngx.var.upstream_status, ngx.var.upstream_response_time)
-  report_req_response_time()
+function _M.log(_, context)
+  local service = context.service.id
+  if not service or not extended_metrics then
+    service = "all"
```

Comment:
Why `all` and not just `nil` ?

Reply:
If I send nil the following error message appears

```
2019/04/30 07:18:03 [error] 149#149: *3 [lua] prometheus.lua:295: log_error(): Wrong number of labels for upstream_status. Expected 2, got 1 while logging request, client: 127.0.0.1, server: _, request: "GET /some-path?user_key=4e5944f11f575fd50bd6624c8d441f13 HTTP/1.1", upstream: "https://18.213.6.196:443/some-path?user_key=4e5944f11f575fd50bd6624c8d441f13", host: "echo-api.3scale.net"
```

Reply:
I see. Might be good to leave there a comment then, so we don't have to ask the question when we revisit that code.

Reply:
And an empty string? I'm not sure what is better honestly.

Reply:
An empty string can be confused, so all have explicit information.